src/leap/bitmask/backend/components.py

```diff
@@ -804,9 +804,19 @@ def register_imap_token(imap_token):
                     self._signaler.soledad_got_service_token,
                     ('imap', imap_token))
 
+        def register_smtp_token(smtp_token):
+            self._service_tokens['smtp'] = smtp_token
+            if self._signaler is not None:
+                self._signaler.signal(
+                    self._signaler.soledad_got_service_token,
+                    ('smtp', smtp_token))
```

Comment:
Maybe we can converge `register_imap_token` and `register_smtp_token` into one function adding a parameter `service` or something to it.